tests/VectorException/enc/init.cpp

```diff
@@ -15,47 +18,62 @@ extern "C" void get_cpuid(
     unsigned int* ecx,
     unsigned int* edx);
 
-static int done = 0;
-static unsigned int c = 0;
-static int hits1 = 0;
-static int hits2 = 0;
+static int done[2];
+static unsigned int c[2];
+static int hits1[2];
+static int hits2[2];
 
 #define AESNI_INSTRUCTIONS 0x02000000u
 
-int test_cpuid_instruction(unsigned int what)
+int test_cpuid_instruction(unsigned int what, int use_exception_handler_stack)
 {
-    if (!done)
+    int index = (use_exception_handler_stack) ? 1 : 0;
+    if (!done[index])
     {
         unsigned int a, b, d;
-        get_cpuid(1, 0, &a, &b, &c, &d);
+        get_cpuid(1, 0, &a, &b, &c[index], &d);
         // Do something with out param so call to cpuid is not optimized out.
         if (a == 0)
         {
             oe_host_printf("This is the value of a: %d", a);
         }
 
         // This should be executed only once.
-        ++hits1;
-        done = 1;
+        ++hits1[index];
+        done[index] = 1;
     }
 
     // This should be executed 3 times.
-    ++hits2;
-    return ((c & what) != 0) ? 1 : 0;
+    ++hits2[index];
+    return ((c[index] & what) != 0) ? 1 : 0;
 }
 
-static int init =
-    (test_cpuid_instruction(500),
-     test_cpuid_instruction(600),
-     test_cpuid_instruction(AESNI_INSTRUCTIONS));
+static int init[2] = {
+    (test_cpuid_instruction(500, 0),
+     test_cpuid_instruction(600, 0),
+     test_cpuid_instruction(AESNI_INSTRUCTIONS, 0)),
+    (test_cpuid_instruction(500, 1),
+     test_cpuid_instruction(600, 1),
+     test_cpuid_instruction(AESNI_INSTRUCTIONS, 1))};
 
-void enc_test_cpuid_in_global_constructors()
+void enc_test_cpuid_in_global_constructors(int use_exception_handler_stack)
 {
-    OE_TEST(init == 1);
-    OE_TEST(done == 1);
-    OE_TEST(c != 0);
-    OE_TEST(hits1 == 1);
-    OE_TEST(hits2 == 3);
+    int index = (use_exception_handler_stack) ? 1 : 0;
+    void* stack;
+
+    stack = malloc(EXCEPTION_HANDLER_STACK_SIZE);
+    if (!stack)
+        return;
+    oe_sgx_set_td_exception_handler_stack(stack, EXCEPTION_HANDLER_STACK_SIZE);
```

Comment:
Not clear what this test is doing.  Between line 67 and 77, no exception is raised; so it doesn't matter whether exception stack is set or not.

Reply:
updated